src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -215,4 +217,74 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         }
         return tagSet;
     }
+    /**
+     * Validates if a {@code String name} is a valid {@code Name}.
+     *
+     * @param input String to validate.
+     * @throws ParseException if the given string is invalid.
+     */
+    public static void validateName(String input) throws ParseException {
+        if (input.isEmpty() || !Name.isValidName(input)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, Name.MESSAGE_CONSTRAINTS));
+        }
+    }
+    /**
+     * Validates if a list of {@code String names} are valid for {@code Name} objects.
+     *
+     * @param inputs List of names to validate.
+     * @throws ParseException if any of the given names are invalid.
+     */
+    public static void validateNames(List<String> inputs) throws ParseException {
+        for (String name : inputs) {
+            validateName(name);
+        }
+    }
```

Comment:
I guess this works but is there a reason its a list of string instead of just a string since you are not going to be keying in a list of names? cause it takes up a little more space?

Reply:
i found a bug in the process of testing this... If i key in find n/avi john(2 separate names) it throws an illegal argument exception and an invocation error you should check out i guess?

But if the purpose is to be able to let you find say multiple persons or multiple tags then i guess it works but you gotta make sure that the user knows how to use that command
eg. i can do `find n/avi n/john` but this was not let known to me 

One more thing is i assume that the find command doesnt stack the filters but in fact implements find 2 times
so if i do `find n/john and fp/insurance` it doesnt find a john with an insurance but finds johns and insurance fps and displays them all

Reply:
Yeah the first one's a bug. I'll fix that.

The format for the updated command has been added to the user guide. Do let me know how the prompt can be improved.

Finding with multiple predicates like an 'or' boolean operator is intended as the original find command also does it like that.